datadog_checks_base/datadog_checks/base/utils/tls.py

```diff
@@ -35,8 +35,8 @@
 class TlsContextWrapper(object):
     __slots__ = ('logger', 'config', 'tls_context')
 
-    def __init__(self, instance, remapper=None):
-        # type: (InstanceType, Dict[AnyStr, Dict[AnyStr, Any]]) -> None
+    def __init__(self, instance, remapper=None, overrides=None):
```

Comment:
There would be different ways to implement this `overrides` param. At first, my understanding based on the doctstring was that any user configuration would be replaced by the content of overrides. i.e the overrides occur on the conf.yaml values directly.

You implemented it at the `TlsContextWrapper.config` level which has a slightly different behavior as the `remapper` has already occured at that point. This is potentially slightly less future-proof: indeed any change to the base class will always have a wide compatibility range with the content of `instance` because of usability. The content of `TlsContextWrapper.config` is internal and could be changed in the future.

Any reason to implement it the way you did? If so let's make sure to explicitly document the drawbacks.

Reply:
Originally I was thinking the `overrides` param would be used for modifying the context itself once it was created, so values like `verify_mode` could be hardcoded to either `ssl.CERT_REQUIRED` or `ssl.CERT_NONE`: https://github.com/DataDog/integrations-core/pull/8268/files#diff-d1d3b592cf2ef4b7512bec5a80f322f0ce4391a524574c9e67693d840adc26bdL309-L311. This way it would be easier to migrate integrations that did custom context creation. Although thinking now about it, I don't think this PR can do that to the extent I wanted.

I implemented at the `TlsContextWrapper.config` level because I originally planned `overrides` specifically for updating legacy integrations. I felt that future integrations with `get_tls_context()` shouldn't really need to use `overrides`, and that they should implement their logic around `get_tls_context()` instead of the other way around? But I think you're right that it would be better to modify the `conf.yaml` values instead in case anything internal is changed.